tensorstore/kvstore/s3/object_metadata_test.cc

```diff
@@ -0,0 +1,27 @@
+// Copyright 2020 The TensorStore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "tensorstore/kvstore/s3/object_metadata.h"
+
+#include <string>
+
+#include <gtest/gtest.h>
+#include "absl/time/time.h"
+#include "tensorstore/kvstore/test_util.h"
+#include "tensorstore/util/result.h"
+#include "tensorstore/util/status.h"
+
+namespace {
+
```

Comment:
:)

Reply:
I'm going to argue that the lack of test cases here has been conveniently side-stepped by the removal of  `ObjectMetadata`.  I swear I will add test cases if `ObjectMetadata` returns (for something like `x-amz-version-id`). :)

`StorageGenerationFromHeaders` is simple as it just wraps the etag header. Let me know if you think there is value in a test case for this.